### PR TITLE
[opencascade] Fix paths in target files for static build

### LINKED ISSUE #31022: [opencascade] paths in target files for static build
### Is your feature request related to a problem? Please describe.

When building with triplet `x64-windows-static`, the paths in target cmake files are `debug/libd`, resulting a build failure by CMake. 

The existing `portfile.cmake` only replaces the paths in dynamic builds. Perhaps we should do it in both dynamic and static?

### Proposed solution

Move the code block https://github.com/microsoft/vcpkg/blob/95252eadd63118201b0d0df0b4360fa613f0de84/ports/opencascade/portfile.cmake#L85-L101 outside the `if (VCPKG_LIBRARY_LINKAGE STREQUAL dynamic)`

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Gellert5225**
Fixes #31022 

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by Gellert5225** [comment]
> @Gellert5225 please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement
> ### Contribution License Agreement
> 
> This Contribution License Agreement (**“Agreement”**) is agreed to by the party signing below (**“You”**), and conveys certain license rights to Microsoft Corporation and its affiliates (“Microsoft”) for Your contributions to Microsoft open source projects. This Agreement is effective as of the latest signature date below.
> 
>     1. **Definitions**.
>        **“Code”** means the computer software code, whether in human-readable or machine-executable form,
>        that is delivered by You to Microsoft under this Agreement.
>        **“Project”** means any of the projects owned or managed by Microsoft and offered under a license
>        approved by the Open Source Initiative ([www.opensource.org](http://www.opensource.org)).
>        **“Submit”** is the act of uploading, submitting, transmitting, or distributing code or other content to any
>        Project, including but not limited to communication on electronic mailing lists, source code control
>        systems, and issue tracking systems that are managed by, or on behalf of, the Project for the purpose of
>        discussing and improving that Project, but excluding communication that is conspicuously marked or
>        otherwise designated in writing by You as “Not a Submission.”
>        **“Submission”** means the Code and any other copyrightable material Submitted by You, including any
>        associated comments and documentation.
> 
>     2. **Your Submission**. You must agree to the terms of this Agreement before making a Submission to any
>        Project. This Agreement covers any and all Submissions that You, now or in the future (except as
>        described in Section 4 below), Submit to any Project.
> 
>     3. **Originality of Work**. You represent that each of Your Submissions is entirely Your original work.
>        Should You wish to Submit materials that are not Your original work, You may Submit them separately
>        to the Project if You (a) retain all copyright and license information that was in the materials as You
>        received them, (b) in the description accompanying Your Submission, include the phrase “Submission
>        containing materials of a third party:” followed by the names of the third party and any licenses or other
>        restrictions of which You are aware, and (c) follow any other instructions in the Project’s written
>        guidelines concerning Submissions.
> 
>     4. **Your Employer**. References to “employer” in this Agreement include Your employer or anyone else
>        for whom You are acting in making Your Submission, e.g. as a contractor, vendor, or agent. If Your
>        Submission is made in the course of Your work for an employer or Your employer has intellectual
>        property rights in Your Submission by contract or applicable law, You must secure permission from Your
>        employer to make the Submission before signing this Agreement. In that case, the term “You” in this
>        Agreement will refer to You and the employer collectively. If You change employers in the future and
>        desire to Submit additional Submissions for the new employer, then You agree to sign a new Agreement
>        and secure permission from the new employer before Submitting those Submissions.
> 
>     5. **Licenses**.
> 
> 
>     * **Copyright License**. You grant Microsoft, and those who receive the Submission directly or
>       indirectly from Microsoft, a perpetual, worldwide, non-exclusive, royalty-free, irrevocable license in the
>       Submission to reproduce, prepare derivative works of, publicly display, publicly perform, and distribute
>       the Submission and such derivative works, and to sublicense any or all of the foregoing rights to third
>       parties.
> 
>     * **Patent License**. You grant Microsoft, and those who receive the Submission directly or
>       indirectly from Microsoft, a perpetual, worldwide, non-exclusive, royalty-free, irrevocable license under
>       Your patent claims that are necessarily infringed by the Submission or the combination of the
>       Submission with the Project to which it was Submitted to make, have made, use, offer to sell, sell and
>       import or otherwise dispose of the Submission alone or with the Project.
> 
>     * **Other Rights Reserved**. Each party reserves all rights not expressly granted in this Agreement.
>       No additional licenses or rights whatsoever (including, without limitation, any implied licenses) are
>       granted by implication, exhaustion, estoppel or otherwise.
> 
> 
>     6. **Representations and Warranties**. You represent that You are legally entitled to grant the above
>        licenses. You represent that each of Your Submissions is entirely Your original work (except as You may
>        have disclosed under Section 3). You represent that You have secured permission from Your employer to
>        make the Submission in cases where Your Submission is made in the course of Your work for Your
>        employer or Your employer has intellectual property rights in Your Submission by contract or applicable
>        law. If You are signing this Agreement on behalf of Your employer, You represent and warrant that You
>        have the necessary authority to bind the listed employer to the obligations contained in this Agreement.
>        You are not expected to provide support for Your Submission, unless You choose to do so. UNLESS
>        REQUIRED BY APPLICABLE LAW OR AGREED TO IN WRITING, AND EXCEPT FOR THE WARRANTIES
>        EXPRESSLY STATED IN SECTIONS 3, 4, AND 6, THE SUBMISSION PROVIDED UNDER THIS AGREEMENT IS
>        PROVIDED WITHOUT WARRANTY OF ANY KIND, INCLUDING, BUT NOT LIMITED TO, ANY WARRANTY OF
>        NONINFRINGEMENT, MERCHANTABILITY, OR FITNESS FOR A PARTICULAR PURPOSE.
> 
>     7. **Notice to Microsoft**. You agree to notify Microsoft in writing of any facts or circumstances of which
>        You later become aware that would make Your representations in this Agreement inaccurate in any
>        respect.
> 
>     8. **Information about Submissions**. You agree that contributions to Projects and information about
>        contributions may be maintained indefinitely and disclosed publicly, including Your name and other
>        information that You submit with Your Submission.
> 
>     9. **Governing Law/Jurisdiction**. This Agreement is governed by the laws of the State of Washington, and
>        the parties consent to exclusive jurisdiction and venue in the federal courts sitting in King County,
>        Washington, unless no federal subject matter jurisdiction exists, in which case the parties consent to
>        exclusive jurisdiction and venue in the Superior Court of King County, Washington. The parties waive all
>        defenses of lack of personal jurisdiction and forum non-conveniens.
> 
>     10. **Entire Agreement/Assignment**. This Agreement is the entire agreement between the parties, and
>         supersedes any and all prior agreements, understandings or communications, written or oral, between
>         the parties relating to the subject matter hereof. This Agreement may be assigned by Microsoft.

@microsoft-github-policy-service agree company="JPPI"

**Comment by Gellert5225** [comment]
@microsoft-github-policy-service agree company="JPPI"

**Comment by dg0yt** [comment]
Can you give more expressive titles? It helps a lot with searching and with e-mail notifications.

**Comment by Gellert5225** [comment]
I got this build error for x86_windows: 

```
##[error]versions/o-/opencascade.json(2,): error : while parsing versions for opencascade from version/o-/opencascade.json
7.6.2#2 is declared with 7350dac73ca1e9c0ca9c2e639651a895683a1966, but the local port has a different SHA d35cfe93781f0d49d78e8a51d20015ae311d7ed7.
Please update the port's version fields and then run:
vcpkg x-add-version opencascade
git add versions
git commit -m "Update version database"
to add the new version.
To attempt to resolve all errors at once, run:
vcpkg x-add-version --all
```

Before I submit the PR I did change the `port-version` from 1 to 2 in the `vcpkg.json`. 

This is my first time contributing, any suggestions?

File: ports/opencascade/portfile.cmake [hunk 1]
```diff
@@ -77,28 +77,28 @@ endif()
...
+)
+
+foreach(TARGET_FILE IN LISTS TARGET_FILES)
+	file(READ "${TARGET_FILE}" filedata)
+	string(REGEX REPLACE "libd" "lib" filedata "${filedata}")
+	string(REGEX REPLACE "bind" "bin" filedata "${filedata}")
+	file(WRITE "${TARGET_FILE}" "${filedata}")
+endforeach()
+
 if (VCPKG_LIBRARY_LINKAGE STREQUAL dynamic)
```
Comment (dg0yt) [inline ports/opencascade/portfile.cmake:98]: Can you add quotes for `"dynamic"` here?

**Review by dg0yt — CHANGES_REQUESTED** [review]
The new block must be guarded with 
~~~cmake
if(NOT VCPKG_BUILD_TYPE)
~~~
to allow for release-only builds.

File: ports/opencascade/portfile.cmake (hunk 1 above)
Comment (LilyWangLL) [reply ports/opencascade/portfile.cmake:98]: Ping @Gellert5225 for response.

Comment (Gellert5225) [reply ports/opencascade/portfile.cmake:98]: Added. Sorry for late response

**Comment by LilyWangLL** [comment]
@Gellert5225 Please modify `port-version` in `vcpkg.json`.

Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". That way, I can be aware that you've responded since you can't modify the tags.

**Comment by LilyWangLL** [comment]
PR #31460 has been merged, so I am closing this PR for now.

**Pull request closed without merging**